jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -435,7 +429,13 @@ public void onPartContent(Content.Chunk chunk)
             if (fileName != null || isUseFilesForPartsWithoutFileName())
             {
                 long maxFileSize = getMaxFileSize();
-                fileSize += buffer.remaining();
+                long fileSize;
+                try (AutoLock ignored = lock.lock())
+                {
+                    fileSize = this.fileSize;
+                    fileSize += buffer.remaining();
+                    this.fileSize = fileSize;
+                }
```

Comment:
I don't think `this.fileSize` needs locking. It is only ever accessed by the parser in the callbacks.
Same for `this.memoryFileSize`.

Reply:
Locking identifies the fields that possibly are subject to concurrency, so helps understanding what are the concurrency entry points.
Having other stuff locked that does not need to seems more confusing to me.

Reply:
Okay, but then `fileSize` and `memoryFileSize` are to be reset outside lock scope in `onPart` for consistency,